core/lib/src/response/named_file.rs

```diff
@@ -38,25 +45,35 @@ impl NamedFile {
         // all of those `seek`s to determine the file size. But, what happens if
         // the file gets changed between now and then?
         let file = File::open(path.as_ref()).await?;
-        Ok(NamedFile(path.as_ref().to_path_buf(), file))
+        Ok(NamedFile {
+            path: path.as_ref().to_path_buf(),
+            file,
+            modified: None,
+        })
+    }
+
+    pub async fn with_last_modified_date<P: AsRef<Path>>(path: P) -> io::Result<NamedFile> {
```

Comment:
A doc comment on this method should explain that it works like `NamedFile::open` and that it sets and checks `Last-Modified` and `If-Modified-Since`, and include an example.

Reply:
Done